cmake/Macros.cmake

```diff
@@ -301,11 +301,39 @@ macro(sfml_add_example target)
 
 endmacro()
 
+# add a new static library which provides a test runner with SFML testing utilities
+# example: sfml_add_test_runner(sfml-test-runner-window
+#                               SOURCES TestUtilities/WindowUtil.hpp TestUtilities/WindowUtil.cpp
+#                               DEPENDS SFML::Window sfml-test-runner-system)
+function(sfml_add_test_runner target)
+
+    # parse the arguments
+    cmake_parse_arguments(THIS "" "" "SOURCES;DEPENDS" ${ARGN})
+    if (NOT "${THIS_UNPARSED_ARGUMENTS}" STREQUAL "")
+        message(FATAL_ERROR "Extra unparsed arguments when calling sfml_add_test_runner: ${THIS_UNPARSED_ARGUMENTS}")
+    endif()
+
+    # set a source group for the source files
+    source_group("" FILES ${THIS_SOURCES})
+
+    # create the target
+    add_library(${target} STATIC ${THIS_SOURCES})
+
+    # set the target's folder (for IDEs that support it, e.g. Visual Studio)
+    set_target_properties(${target} PROPERTIES FOLDER "Tests/Runners")
+
+    # link the target to its SFML dependencies and its test runner library
+    target_link_libraries(${target} PUBLIC ${THIS_DEPENDS})
+
+    set_target_warnings(${target})
+endfunction()
+
 # add a new target which is a SFML test
 # example: sfml_add_test(sfml-test
-#                           ftp.cpp ...
-#                           SFML::Network)
-function(sfml_add_test target SOURCES DEPENDS)
+#                        ftp.cpp ...
+#                        SFML::Network
+#                        sfml-test-runner-system)     # Library providing the test runner and related utilities
+function(sfml_add_test target SOURCES DEPENDS TEST_RUNNER)
```

Comment:
Where does the "runner" terminology come from? To me that connotes CI runners, the machines that execute our CI jobs. Could we change that to some term that reflects that each "runner" is in fact a helper library?

Reply:
It's "test runner", which I think is quite an established term.

Not sure what else you could use... "test engine", "test framework" etc. seem also a bit out of place, as we're really running them and not providing the underlying infrastructure.

Maybe just "test" or "tests"?

Reply:
That comes from vague memories of documentation I skimmed through. I am not sure if it is the best word.
After some research, [boost.Test documentation](https://www.boost.org/doc/libs/1_79_0/libs/test/doc/html/boost_test/adv_scenarios/test_module_runner_overview.html) seems to agree. It also gives alternatives "orchestrator" and "driver".
I am open to suggestion.

Reply:
> Not sure what else you could use... "test engine", "test framework" etc. seem also a bit out of place, as we're really running them and not providing the underlying infrastructure.

I'd name each utils library `sfml-${module}-test-utils` because it's just some test utilities. It's not running anything. It's just some compiled C++ code with some headers for using the code. Each one of those utils library provides a header called `{Module}Utils.hpp` so it's all very consistent this way.

Reply:
> It's not running anything.

That is not true. It comes with a main function from DoctestMain.cpp which does all the magic of command line parsing, test running and console output.
Still `sfml-${module}-test-utils` sounds more like a library than `sfml-test-runner-${module}` so that could be make more sense.

Reply:
> It comes with a main function from DoctestMain.cpp

I'm aware and I still disagree that it's "running" the code so to speak. Sometimes libraries implement `main()`. Non-header-only test frameworks like Catch v3 and GTest do this. Can we at least name it `sfml-${module}-test-runner` since that reads better than `sfml-test-runner-${module}`?

Reply:
I changed names to `sfml-${module}-test-utils`. I am still not sure what is best so do not hesitate to ask for something else.